File: python/space-age/space_age.py
class SpaceAge:
    def __init__(self, seconds):
        self.age = seconds / 31557600

    def on_earth(self):
        return round(self.age, 2)

    def on_mercury(self):
        return round(self.age / 0.2408467, 2)
                
    def on_venus(self):
        return round(self.age / 0.61519726, 2)
        
    def on_mars(self):
        return round(self.age / 1.8808158, 2)
        
    def on_jupiter(self):
        return round(self.age / 11.862615, 2)
        
    def on_saturn(self):
        return round(self.age / 29.447498, 2)
        
    def on_uranus(self):
        return round(self.age / 84.016846, 2)
        
    def on_neptune(self):
        return round(self.age / 164.79132, 2)

File: ruby/space-age/space_age.rb
class SpaceAge
  attr_accessor :age
  
  def initialize(seconds)
    @age = seconds / 31557600.0
  end

  def on_earth
    @age
  end

  def on_mercury
    @age / 0.2408467
  end

  def on_venus
    @age / 0.61519726
  end

  def on_mars
    @age / 1.8808158
  end

  def on_jupiter
    @age / 11.862615
  end

  def on_saturn
    @age / 29.447498
  end

  def on_uranus
    @age / 84.016846
  end

  def on_neptune
    @age / 164.79132
  end
end
